server/quota/quota.go

```diff
@@ -0,0 +1,117 @@
+package quota
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/config"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"github.com/tigrisdata/tigris/server/transaction"
+	"go.uber.org/atomic"
+	"golang.org/x/time/rate"
+)
+
+var (
+	sizeLimitUpdateInterval int64 = 5 // seconds
+
+	ErrRateExceeded        = api.Errorf(api.Code_RESOURCE_EXHAUSTED, "request rate limit exceeded")
+	ErrThroughputExceeded  = api.Errorf(api.Code_RESOURCE_EXHAUSTED, "request throughput limit exceeded")
+	ErrStorageSizeExceeded = api.Errorf(api.Code_RESOURCE_EXHAUSTED, "data size limit exceeded")
+)
+
+type State struct {
+	Rate         *rate.Limiter
+	Throughput   *rate.Limiter
+	Size         atomic.Int64
+	SizeUpdateAt atomic.Int64
+	SizeLock     sync.Mutex
+}
+
+type Manager struct {
+	quota     sync.Map
```

Comment:
should we call it `tenantQuota` to make it clear that the tenant state is managed on this map.

Reply:
quota.tenantQuota doesn't sound right to me.
quota.Manager is better.

Reply:
no, I mean for the map,`quota sync.Map`. The quota.Manager is fine.